### ISSUE TITLE
add a prometheus--text-file-exporter parameter

### ISSUE THREAD

**Issue by Stunkymonkey** [borgbackup/borg#7911]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Feature request

## Describe the problem you're observing.

when running borg it is only possible to parse the output from `--stats` to get a more deeper look into metrics.

It would be nice to have an parameter `--prometheus-exporter <file>` to write all the metrics to the file. This way we do not need to run any additional command and have it persistent on the disk.

**Comment by ThomasWaldmann** [comment]
Some commands also have json output capability, so maybe you can use that.

**Comment by Stunkymonkey** [comment]
That is a good idea. Maybe I can use the [json-exporter](https://github.com/prometheus-community/json_exporter ) for this. Not sure how good the metrics are, but I will have a look.

Does that mean you don't want to provide it in the first place, but would rather suggest the use of other tools?

**Comment by ThomasWaldmann** [comment]
Yes, guess we rather provide generic exporting (like json) than supporting misc. more specialised formats (? no idea about what prometheus need) directly.

**Comment by Stunkymonkey** [comment]
I am very fine with json, but could we add an parameter like `--json-export file.json` so I have the normal output of the stats ind the job and my monitoring will use the json.

**Comment by ThomasWaldmann** [comment]
IIRC, if you use `--log-json`, you'll get all log output as json.

Some commands have a `--json` (or `--json-lines`) option, so their (non-logging) output is made in json format.

**Comment by Stunkymonkey** [comment]
yes, but then I would have either the stats in a nice format or json, but never both.

If you do not like my suggestion, please go ahead an close it. I am ok with the current situation.

**Comment by securitym0nkey** [comment]
I would like to see this as a feature. And the openmetrics format has become the de-facto industry standard.
